File: UI/EStore.UI/src/app/common/ProductDto.ts
import { Category } from "./Category";

export class ProductDto {
    productId: number;
    name: string;
    price: number;
    description: string;
    categoryName: string;
    imageUrl: string;
    quantity: number;
    rating: number;
    categoryId: number;
    category: Category;
}

File: Services/EStore.Service.ProductApi/Models/Dto/ProductDto.cs
﻿using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace EStore.Service.ProductApi.Models.Dto
{
	public class ProductDto
	{
		public int ProductId { get; set; }
		public string Name { get; set; }
		public double Price { get; set; }
		public string Description { get; set; }
		public string CategoryName { get; set; }
		public string ImageUrl { get; set; }
		[JsonPropertyName("quantity")]
		public int StockQuantity { get; set; }
		public int CategoryId { get; set; }
		public CategoryDto Category { get; set; }

	}
}
